book/course/conda.md

```diff
@@ -0,0 +1,540 @@
+
+```bash
+$ conda env create -f environment.yml
+```
+
+Here we're specifying Conda create a new environment and using the `-f` option to specify that it creates the environment using a file with an environment specification.
+We pass the file path to the environment file as the argument following `-f`.
+
+#### Creating a cross platform environment file
+
+As noted above using `conda env export` creates a highly specific environment file, this often causes difficulties when sharing environments across operating systems as the `environment.yml` contains operating system specific hashes for each package.
+
+There are two possible methods of creating a more flexible `environment.yml`.
+
+##### 1. Using `conda env export --from-history`
+
+By default `conda env export` exports an environments entire specification, including dependencies of packages you `conda install` and their associated hashes. 
+If you use `conda env export --from-history` Conda only exports packages explicitly installed with `conda install`. 
+It does not include dependencies of those packages and therefore allows different operating systems to more flexibly install package dependencies.
+
+For the above example with `data-sci-env` we would export a more flexible `environment.yml` with:
+```bash
+(data-sci-env)$ conda env export --from-history | grep -v ^prefix: > environment.yml
+```
+
+##### 2. Manually create an `environment.yml`
+
+The other option is to manually specify the `environment.yml` file. 
+This is often more fiddly than just exporting an environment but can be preferable to ensure all the desired dependencies of your project are captured.
+Environment files are written in YAML, a markup language, and have the standard pattern of:
+```yaml
+name: data-sci-env
+channels:
+- defaults
+dependencies:
+- scikit-learn
+- matplotlib=3.5.1
+- pandas=1.4.3
+```
+Where you specify the environment name, a list of Conda channels used to install packages, and under dependencies a list of packages to be installed. You can also include version specification within the `environment.yml` allowing you to 
+
+Understanding the differences between weays to create environment files is important when you come to deciding on how best to share your project. 
+It's important to consider the balance of reproducibility and portability, `conda env export` captures the exact specification of an environment including all installed packages, their dependencies and package hashes.
+Sometimes this level of detail should be included to ensure maximum reproduciblity of a project, when looking to validate results, but it's important to also balance being able to allow people to reproduce your work on other systems.
+
+
+## Using conda to install packages
+
+With the Conda command line tool searching for and installing packages is can be performed with the following subcommands:
+- `conda search`
+- `conda install`
+
+### Searching for packages
+
+```bash
+$ conda search python
+```
+```output
```

Comment:
I'm not sure I understand the consistency between sometimes splitting and other times not the output.  I was tempted by toggleable sections, which I might query you about later...